### ISSUE TITLE
nested modules 

### ISSUE THREAD

**Issue by shmaram** [sascha245/vuex-simple#16]
-  **I'm submitting a ...**
[ ] bug report
[ ] feature request
[ ] question about the decisions made in the repository
[ X ] question about how to use this project

-  **Summary**

I would like to know if there is a way to have a nested modules, as you can find here : https://vuex.vuejs.org/guide/modules.html

*  **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

**Comment by sascha245** [comment]
Hello @shmaram,

By default, all modules loaded with the StoreBuilder will be nested in the store once you create it with `storeBuilder.create()`.
So you could theoretically access the state of the `CounterModule` (see readme) with `store.state.counter`.

However, note that it is currently not possible to nest modules in other modules.

Hope that answers your question,

Sascha

**Comment by shmaram** [comment]
Hi
The current Vuex version does support nesting of module using the namespace option.
Can you tell if this will be supported also in vuex-simple?

**Comment by sascha245** [comment]
Hi @shmaram,

All modules in the store are namespaced by the string given in the `@Module(<namespace>)` decorator.
If we take the `CounterModule` example again, we can theorically call the increment mutation with `store.commit('counter/increment')`.

Concerning nesting modules in other modules, for now it is not planned to be implemented, as I am currently still working on another library I need. Once that library is done however, I wouldn't mind working on it if people are really interested in this feature.

I wish you a good evening,

Sascha

**Comment by shmaram** [comment]
Hi @sascha245 
Thanks for the answer. 
The problem we are facing is that we have a very long module list and at the moment due to it been is flat list it is very hard to keep track of the vuex structure. 
It would be very helpful for us if you could support this option in this package.

Shahar

**Comment by sonsina** [comment]
Hi @sascha245 ,
Actually @shmaram 's timing is really good since I am looking for this option as well. I'd be happy to see this developed, hopefully soon.

**Comment by sascha245** [comment]
So, I did manage to add submodules, but without making breaking changes it makes a few things quite messy and more complicated than it should be.

That is why I will probably make a new major version soon. More details can be found under #17.
Don't hesitate to tell me what you think about it and if you have any other suggestions.

I wish you a good evening,

Sascha

**Comment by shmaram** [comment]
Thank you for this update! 👍  
I have commented my thoughts in the 2.0.0 thread

**Comment by sascha245** [comment]
Hi @shmaram and @sonsina,

Submodules are now available in the new major version 2.0.0. 
A recap about the changes is available in #17. The README has also been updated and should contain most information about how to easily migrate to the new version.

Sascha

**Comment by sonsina** [comment]
Hi @sascha245 , thanks a lot for the development. I had the time to integrate it and it is working really great.

**Comment by sascha245** [comment]
That is nice to hear! And in the case you find a bug or think there is another important feature that is missing, don't hesitate to open up a new ticket.
